src/celeritas/global/ActionInterface.hh

```diff
@@ -56,6 +56,11 @@ class ActionInterface
   public:
     // Default virtual destructor allows deletion by pointer-to-interface
     virtual ~ActionInterface();
+    ActionInterface() = default;
+    ActionInterface(ActionInterface const&) = delete;
+    ActionInterface& operator=(ActionInterface const&) = delete;
+    ActionInterface(ActionInterface&&) = delete;
+    ActionInterface& operator=(ActionInterface&&) = delete;
```

Comment:
Same comment about pure virtual base classes. There's even a note in the core guidelines:
> Writing these functions can be error-prone.